lib/internal/fs/streams.js

```diff
@@ -386,22 +386,50 @@ WriteStream.prototype.open = openWriteFs;
 
 WriteStream.prototype._construct = _construct;
 
+function writeAll(data, pos, cb) {
+  const len = data.length;
+  this[kFs].write(this.fd, data, 0, len, pos, (er, bytesWritten) => {
+    if (this.destroyed || er) {
+      return cb(er);
+    }
+
```

Comment:
Do we need some kind of guard to avoid infinite retries?

Reply:
I think so, maybe 5 times hardcoded?